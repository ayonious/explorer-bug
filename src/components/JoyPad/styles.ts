import styled from 'styled-components';

export const Row = styled.div`
  display: flex;
  justify-content: center;
  width: fit-content;
  margin: auto;
`;
