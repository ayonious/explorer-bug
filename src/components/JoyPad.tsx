import * as React from 'react';
import { Row } from '../styledcomponents/styles';
import { CustomButton } from './Utils';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

interface State {}

export default class JoyPad extends React.Component<Props, State> {
  render() {
    return (
      <div>
        <Row>
          <CustomButton onClick={this.props.handleGo} title={'Go'} />
        </Row>
        <Row>
          <CustomButton onClick={this.props.handleLeft} title={'Left'} />
          <CustomButton onClick={this.props.handleRight} title={'Right'} />
        </Row>
      </div>
    );
  }
}
