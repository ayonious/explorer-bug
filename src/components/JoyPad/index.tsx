import * as React from 'react';

import { Row } from './styles';
import CustomButton from '../CustomButton';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

const JoyPad = (props: Props) => {
  return (
    <div>
      <Row>
        <CustomButton onClick={props.handleGo} title={'Go'} />
      </Row>
      <Row>
        <CustomButton onClick={props.handleLeft} title={'Left'} />
        <CustomButton onClick={props.handleRight} title={'Right'} />
      </Row>
    </div>
  );
};

export default JoyPad;
