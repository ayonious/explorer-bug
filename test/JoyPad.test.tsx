/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import { expect } from '@jest/globals';
import React from 'react';

import JoyPad from '../src/components/JoyPad';

it('JoyPad renders', () => {
  const func = () => {};
  const { container } = render(
    <JoyPad handleGo={func} handleLeft={func} handleRight={func} />
  );
  expect(container).toMatchSnapshot();
});
