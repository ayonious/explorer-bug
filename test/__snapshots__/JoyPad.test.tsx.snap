// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoyPad renders 1`] = `
<div>
  <div>
    <div
      class="sc-bdnxRM hpWqnB"
    >
      <div
        class="sc-gtsrHT loDeyg"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Go
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="sc-bdnxRM hpWqnB"
    >
      <div
        class="sc-gtsrHT loDeyg"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Left
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-gtsrHT loDeyg"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Right
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
